src/ecalc/libraries/libecalc/common/libecalc/common/utils/rates.py

```diff
@@ -660,6 +670,18 @@ def for_period(self, period: Period) -> Self:
             typ=self.typ,
         )
 
+    def for_timestep(self, current_timestep: datetime) -> Self:
+        timestep_index = next(
+            timestep_index for timestep_index, timestep in enumerate(self.timesteps) if timestep == current_timestep
+        )
+        return self.__class__(
+            timesteps=self.timesteps[timestep_index : timestep_index + 1],
+            values=self.values[timestep_index : timestep_index + 1],
+            regularity=self.regularity[timestep_index : timestep_index + 1],  # type: ignore
```

Comment:
Why ignore type?

Reply:
copy paste...idk :p i think it was a bug where i copied from, that fixed it easily with ignoring it..